.circleci/config.yml

```diff
@@ -0,0 +1,59 @@
+version: 2.1
+
+references:
+  images:
+    go: &GOLANG_IMAGE circleci/golang:latest
+  environments:
+    tmp: &TEST_RESULTS_PATH /tmp/test-results # path to where test results are saved
+
+# reusable 'executor' object for jobs
+executors:
+  go:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      - TEST_RESULTS: *TEST_RESULTS_PATH
+
+jobs:
+  go-test:
+    executor: go
+    steps:
+      - checkout
+      - run: mkdir -p $TEST_RESULTS
+
+      - restore_cache: # restore cache from dev-build job
```

Comment:
Uber-nit: `dev-build` doesn't seem to be a job in this config, this comment is confusing.

Reply:
Thanks! QQ: I'm setting up the project in circle: https://circleci.hashicorp.engineering/setup-project/gh/hashicorp/go-retryablehttp and by default, it builds on every branch. So after the setup and a commit, it should do a fresh build. Any other steps I'm missing?

Reply:
@mdeggies for some reason, CircleCI won't build anything until there is some valid CircleCI config on master, so I usually just push a very minimal config to master without a PR just to get it building first.

Reply:
Thanks guys.